### PR TITLE
TuneUp menu item should be under Extensions menu.

### PR DESCRIPTION
**Pull request by reddyashish**
When TuneUp package is installed, "Show TuneUp" menu item should be present under the Extensions menu.

To use the AddExtensionMenuItem API from ViewLoadedParams class, the DynamoCore references are pointed to the Dynamo 2.12 version. The TargetFrameworkVersion is also set to v4.8.

Screen Shot 2021-07-23 at 8 44 03 AM

### CODE REVIEW

File: TuneUp/TuneUpViewExtension.cs [hunk 1]
```diff
@@ -49,7 +49,7 @@ public void Loaded(ViewLoadedParams p)
                 }
 
             };
-            p.AddMenuItem(MenuBarType.View, TuneUpMenuItem);
+            p.AddExtensionMenuItem(TuneUpMenuItem);
```
Comment (QilongTang) [inline TuneUp/TuneUpViewExtension.cs:52]: hmm can you test loading this view ex in legacy version of Dynamo? I think part of the AC is that if user download newer version of TuneUp in older version of Dynamo e.g. 2.6.1, Dynamo should not crash when loading TuneUp. This code may crash in older version of Dynamo, we may need to wrap this code in try catch and fall back to the older version of API

**Comment by reddyashish** [comment]
@QilongTang After discussing with @mjkkirschner, using Dynamic ViewLoadedParams object, we can call the respective API's. Tested it on latest master as well as older versions of Dynamo. 

**Latest master:**
Screen Shot 2021-07-30 at 1 48 33 PM



**2.11 and older versions of Dynamo:**
Screen Shot 2021-07-30 at 1 48 41 PM

File: TuneUp/TuneUpViewExtension.cs (hunk 1 above)
Comment (reddyashish) [reply TuneUp/TuneUpViewExtension.cs]: Address this issue in the latest commit.

File: TuneUp/TuneUpViewExtension.cs [hunk 2]
```diff
@@ -9,24 +9,26 @@ namespace TuneUp
...
 
-        public void Startup(ViewStartupParams p)
+        public override void Startup(ViewStartupParams p)
         {
         }
 
-        public void Loaded(ViewLoadedParams p)
+        public override void Loaded(ViewLoadedParams p)
         {
+            dynamic dp = (dynamic) p;
```
Comment (QilongTang) [inline TuneUp/TuneUpViewExtension.cs:29]: Can you add comments to this?

**Review by QilongTang — APPROVED** [review]
One comment then LGTM

**Comment by reddyashish** [comment]
Addressed all comments. Merging this.

**Pull request merged**